src/efi.rs

```diff
@@ -97,29 +76,29 @@ impl Efi {
         return esp_device;
     }
 
-    pub(crate) fn ensure_mounted_esp(&self, root: &Path) -> Result<PathBuf> {
-        let mut mountpoint = self.mountpoint.borrow_mut();
-        if let Some(mountpoint) = mountpoint.as_deref() {
-            return Ok(mountpoint.to_owned());
-        }
-        for &mnt in ESP_MOUNTS {
-            let mnt = root.join(mnt);
-            if !mnt.exists() {
-                continue;
-            }
-            let st =
-                rustix::fs::statfs(&mnt).with_context(|| format!("statfs failed for {mnt:?}"))?;
-            if st.f_type != libc::MSDOS_SUPER_MAGIC {
-                continue;
-            }
-            util::ensure_writable_mount(&mnt)?;
-            log::debug!("Reusing existing {mnt:?}");
-            return Ok(mnt);
+    // Get mounted point for esp
+    pub(crate) fn get_mounted_esp(&self, root: &Path) -> Result<Option<PathBuf>> {
+        // First check all potential mount points without holding the borrow
+        let found_mount = ESP_MOUNTS.iter().map(|&mnt| root.join(mnt)).find(|mnt| {
+            mnt.exists()
+                && rustix::fs::statfs(mnt).map_or(false, |st| st.f_type == libc::MSDOS_SUPER_MAGIC)
+                && util::ensure_writable_mount(mnt).is_ok()
```

Comment:
This "swallows" errors. Instead I think we should use `try_fold`; see e.g. this thread https://github.com/bootc-dev/bootc/pull/1299#discussion_r2087544274

Reply:
Updated, could you help to review? Thanks!

Reply:
It looks better yes. However now that I look even more closely I think it'd likely be simpler to just go back to explicit iteration? But OK as is too.